### PR TITLE
[cmake] Create OpenSim/Testing/resources directory.

### PR DESCRIPTION
**Pull request by chrisdembia**
I created a small cmake mechanism that allows us to keep testing resources in
one place, and reduce the duplication of same/similar model files (or other
files) across the repository.

Now, the idea is to store files that multiple tests would want to use in a
single directory. To facilitate this, I created a CMake function
OPENSIM_COPY_TEST_RESOURCES that copies files from this resources dir in the
repository into the correct binary dir.

I altered the python wrapping tests so that they can benefit from this mechanism.

@aseth1 could you review this?

### CODE REVIEW

File: cmake/OpenSimMacros.cmake [hunk 1]
```diff
@@ -143,3 +143,21 @@ FUNCTION(OPENSIM_ADD_LIBRARY)
...
+
+# Copy a file from the directory containing test resources (model files, data,
+# etc.) to the directory in which a test will be executed. This function makes
+# it easy to re-use files that are used in tests. With an easier mechanism for
+# re-using resources, we won't end up version-controllingthe same file into
+# multiple test directories.
+#
+# Arguments are a list of files in the test resources directory
+# (OPENSIM_TEST_RESOURCES_DIR) to copy.
+FUNCTION(OPENSIM_COPY_TEST_RESOURCES)
```
Comment (sherm1) [inline cmake/OpenSimMacros.cmake]: Maybe use a "less screamy" capitalization convention for functions? I think the CMake folks use something like `OpenSimCopyTestResources` for their functions.

**Comment by sherm1** [comment]
Simbody has a similar concept for files shared among its examples. But there the shared directory is called `shared` rather than `resources`. It seemed more important to emphasize the sharing aspect; the fact that they are resources is kind of obvious I guess. Linux has a similar convention with `/usr/share` and `/usr/local/share`. How about switching to `shared` here also?

File: cmake/OpenSimMacros.cmake (hunk 1 above)
Comment (chrisdembia) [inline cmake/OpenSimMacros.cmake]: What do you think about `opensim_copy_test_resources`? I think function names are case insensitive. I still plan to descreamify the entire repo; is it okay if I keep screamy consistency across the repo for now, and descreamify separately?

However, using camelcase also helps distinguish real cmake methods from ones we've made (though I've tried to do this by putting OpenSim in the name of the function).

**Comment by chrisdembia** [comment]
> How about switching to shared here also?

I think that's a good idea. I was partly trying to avoid potential confusion with "shared libraries." However if I name the function `opensim_copy_shared_test_files`, I don't think that is easily confused with shared libraries. So yeah I'll make that change.

**Comment by aymanhab** [comment]
@chrisdembia I like the idea but we need to verify that the actual contents of the files being shared are the same not only file names. It's hard to tell if this was done already or is the plan to introduce the mechanism and then have tests migrate gradually to the new "shared" mechanism.

**Comment by chrisdembia** [comment]
@aymanhab I think a gradual "from here on out" migration is more practical. The cost to our repo size has already been incurred, anyway, so I don't know if we really need to start moving a bunch of model files from existing test dirs.

With this PR I did not alter the inputs to any C++ tests, so there is no issue of same-name-different-contents for C++ tests. However I might have changed the contents of the inputs used for the python tests. I am reasonably confident that this does not affect the python tests though.

I neglected to commit the moved model files, so I am going to re-make this PR after the review is complete. Please review the PR here. When the review is complete, I'll make a new PR that we can hopefully merge quickly.

File: cmake/OpenSimMacros.cmake (hunk 1 above)
Comment (sherm1) [inline cmake/OpenSimMacros.cmake]: > What do you think about opensim_copy_test_resources

Some convention different from variable names would be nice. I'm happy with whatever you think makes sense.

Comment (sherm1) [inline cmake/OpenSimMacros.cmake]: Maybe @elen4 has some advice (on capitalization conventions for CMake function names)?

**Pull request closed without merging**